planning/behavior_velocity_planner/src/scene_module/out_of_lane/scene_out_of_lane.cpp

```diff
@@ -0,0 +1,231 @@
+// Copyright 2023 TIER IV, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "scene_module/out_of_lane/scene_out_of_lane.hpp"
+
+#include "scene_module/out_of_lane/debug.hpp"
+#include "scene_module/out_of_lane/decisions.hpp"
+#include "scene_module/out_of_lane/footprint.hpp"
+#include "scene_module/out_of_lane/lanelets_selection.hpp"
+#include "scene_module/out_of_lane/overlapping_range.hpp"
+#include "scene_module/out_of_lane/types.hpp"
+
+#include <lanelet2_extension/utility/query.hpp>
+#include <lanelet2_extension/utility/utilities.hpp>
+#include <tier4_autoware_utils/system/stop_watch.hpp>
+#include <utilization/debug.hpp>
+#include <utilization/util.hpp>
+
+#include <lanelet2_core/geometry/LaneletMap.h>
+
+#include <memory>
+#include <string>
+#include <vector>
+
+namespace behavior_velocity_planner::out_of_lane
+{
+
+using visualization_msgs::msg::Marker;
+using visualization_msgs::msg::MarkerArray;
+
+OutOfLaneModule::OutOfLaneModule(
+  const int64_t module_id, PlannerParam planner_param, const rclcpp::Logger & logger,
+  const rclcpp::Clock::SharedPtr clock)
+: SceneModuleInterface(module_id, logger, clock), params_(std::move(planner_param))
+{
+}
+
+bool OutOfLaneModule::modifyPathVelocity(
+  PathWithLaneId * path, [[maybe_unused]] StopReason * stop_reason)
```

Comment:
If posibke, would you support the stop reason? It will be used as a high-level API.

Reply:
I now support the `StopReason` as well as the `VelocityFactor`.